File: Pythonleetcode/Leetcode1431.py
class Solution(object):
    def kidsWithCandies(self, candies, extraCandies):
        ans=[]
        candies_sort=sorted(candies)
        for i in range(len(candies)):
            if candies[i]+extraCandies>=candies_sort[-1]:
                ans.append(True)
            else:
                ans.append(False)


        return ans






print(Solution().kidsWithCandies([4,2,1,1,2], 1))

print(Solution().kidsWithCandies([12,1,12],  10))

File: CppPractice/Project1/Project1/Leetcode1431.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

using namespace std;

class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int max_val = *max_element(candies.begin(), candies.end());
        vector<bool> ans;
        for (int i=0; i < candies.size();i++) {
            if (candies[i]+extraCandies>=max_val) {
                ans.push_back(true);
            
            }
            else { ans.push_back(false); }

        }
        return ans;
    }
};
